pepper-apis/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -505,6 +519,83 @@ private static void setupTaskReceiver(Config cfg, String projectId) {
+        GoogleCredentials credentials = GoogleCredentialUtil.initCredentials(ensureDefault);
+        if (credentials == null) {
+            throw new DDPException("Could not get bucket credentials");
+        }
+
+        var storageClient = new GoogleBucketClient(projectId, credentials);
+        var receiver = new FileScanResultReceiver(storageClient,
+                cfg.getString(ConfigFile.FileUploads.UPLOADS_BUCKET),
+                cfg.getString(ConfigFile.FileUploads.SCANNED_BUCKET),
+                cfg.getString(ConfigFile.FileUploads.QUARANTINE_BUCKET));
+
+        // This is the default thread factory, just setting a custom name here.
+        ThreadFactory threadFactory = runnable -> {
+            Thread thread = new Thread(runnable);
+            thread.setName("file-scan-result-subscriber-1");
+            thread.setDaemon(true);
+            return thread;
+        };
+
+        // For now, handle messages one at a time so we don't overwhelm resources.
+        ExecutorProvider executorProvider = InstantiatingExecutorProvider.newBuilder()
+                .setThreadFactory(threadFactory)
+                .setExecutorThreadCount(1)
+                .build();
+
+        ExecutorService callbackExecutor = Executors.newSingleThreadExecutor();
+        Subscriber.Builder builder = Subscriber.newBuilder(subName, receiver)
+                .setSystemExecutorProvider(executorProvider)
+                .setExecutorProvider(executorProvider)
+                .setParallelPullCount(1);
+
+        startNewSubscriberWithRecovery(subName, builder, executorProvider, callbackExecutor, subscriber -> {
+            LOG.info("Started file scan result subscriber to subscription {}", subName);
+            fileScanResultSubscriber = subscriber;
+        });
+    }
+
+    private static void startNewSubscriberWithRecovery(ProjectSubscriptionName subscription,
+                                                       Subscriber.Builder builder,
+                                                       ExecutorProvider executorProvider,
+                                                       ExecutorService callbackExecutor,
+                                                       Consumer<Subscriber> consumer) {
+        Subscriber newSubscriber = builder.build();
+
+        // Add handler that recursively calls back in here to renew subscriber.
+        newSubscriber.addListener(new ApiService.Listener() {
+            @Override
+            public void failed(ApiService.State from, Throwable failure) {
+                LOG.error("Subscriber to subscription {} encountered unrecoverable failure from {}"
+                        + ", rebuilding subscriber", subscription, from, failure);
+                if (!executorProvider.getExecutor().isShutdown()) {
+                    startNewSubscriberWithRecovery(subscription, builder, executorProvider, callbackExecutor, consumer);
```

Comment:
@zyme @evgeniipr I hope I'm setting up this recovery correctly... Advice here will be appreciated!

Reply:
May be to add this line h (before throw `new DDPException`:
```
} catch (TimeoutException timeoutException) {
      subscriber.stopAsync();
}
```

Reply:
Looks like in this case will be called a signature `void error(String var1, Object... var2);`
```
LOG.error("Subscriber to subscription {} encountered unrecoverable failure from {}"
                        + ", rebuilding subscriber", subscription, from, failure);
```

Reply:
@evgeniipr I'm curious why we need to call `stopAsync()`. If subscriber wasn't running already after waiting for the timeout, then do we need to call stop? It might be a good extra safeguard but just wasn't sure if we need it.

Reply:
I'm not sure if it really needs to call (may be really extra safeguard?). At least in this example it's called: https://cloud.google.com/pubsub/docs/samples/pubsub-subscriber-error-listener

Reply:
In that code sample, they're demonstrating letting the subscriber run for 30 seconds and then shutting it down. The way they achieve the shutdown is using `awaitTerminated()` as a mechanism to throw a `TimeoutException` after 30 seconds, and then calling `stopAsync()` on timeout. I'm not sure that has anything to do with starting up the subscriber though...